metadata-ws/src/test/resources/features/release-date.feature

```diff
@@ -0,0 +1,174 @@
+    And the response code should be 201
+    When I request PATCH STUDY1 with list PUBLICATION1 of publications
+    Then the response code should be 2xx
+
+    # Create a web resource for Study2
+    When I request POST /webResources with JSON payload:
+    """
+    {
+      "type": "CENTER_WEB",
+      "resourceUrl": "http://test-release-date.example.com"
+    }
+    """
+    Then set the URL to WEBRESOURCE2
+    And the response code should be 201
+    When I request PATCH STUDY2 with list WEBRESOURCE2 of resources
+    Then the response code should be 2xx
+
+    # Set both studies to released initially
+    When I request PATCH STUDY1 with patch and day null
+    Then the response code should be 200
+    When I request PATCH STUDY1 with patch and day yesterday
+    Then the response code should be 200
+
+    # Make sure that all elements are available initially
+    When I request GET with value of ANALYSIS1
+    Then the response code should be 2xx
+    When I request GET with value of ANALYSIS2
+    Then the response code should be 2xx
+    When I request GET with value of FILE1
+    Then the response code should be 2xx
+    When I request GET with value of FILE2
+    Then the response code should be 2xx
+    When I request GET with value of FILE3
+    Then the response code should be 2xx
+    When I request GET with value of PUBLICATION1
+    Then the response code should be 2xx
+    When I request GET with value of WEBRESOURCE2
+    Then the response code should be 2xx
+
+    # Set release dates for studies according to the scenario
+    When I request PATCH STUDY1 with patch and day <S1_RELEASE>
+    Then the response code should be 200
+    When I request PATCH STUDY2 with patch and day <S2_RELEASE>
+    Then the response code should be 200
+
+    # Check availability of entities according to the scenario
+    When I request GET with value of ANALYSIS1
+    Then the response code should be <A1>
+    When I request GET with value of ANALYSIS2
+    Then the response code should be <A2>
+    # Check availability of files according to the scenario
+    When I request GET with value of FILE1
+    Then the response code should be <F1>
+    When I request GET with value of FILE2
+    Then the response code should be <F2>
+    When I request GET with value of FILE3
+    Then the response code should be <F3>
+    When I request GET with value of PUBLICATION1
+    Then the response code should be <P1>
+    When I request GET with value of WEBRESOURCE2
+    Then the response code should be <W2>
+
+    Examples:
+      # Analysis, Publication, and WebResource are available WHEN AND ONLY WHEN their parent study is available
+      # File is available WHEN AND ONLY WHEN at least one study which links to it is available
+      # Release dates of "null" or "yesterday" should result in a RELEASED state;
+      # "tomorrow" in an UNRELEASED state.
+      | S1_RELEASE | S2_RELEASE | A1  | A2  | F1  | F2  | F3  | P1  | W2  |
```

Comment:
sorry can this be done for today as well ?

Reply:
No problem at all, added `today` to list of checked values. Now every combination of `none`, `yesterday`, `today` and `tomorrow` is tested. Fixed in https://github.com/EBIvariation/metadata-ws/pull/53/commits/588c05d0d135683d056d38b461f0194e486c185d.